ethereum/contracts/Bridge.sol

```diff
@@ -79,29 +82,48 @@ contract Bridge is IBridge, AccessControl {
         _;
     }
 
-    /* Only callable by owner or delegate */
-    modifier onlyDelegate() {
-        require(hasRole(DELEGATE_ROLE, msg.sender) || _isOwner(), "Caller must be owner or delegate");
+    modifier onlyRelayer() {
+        require(hasRole(RELAYER_ROLE, msg.sender), "Caller must be relayer");
         _;
     }
 
-    modifier onlyRelayer() {
-        require(hasRole(RELAYER_ROLE, msg.sender), "Caller must be relayer");
+    modifier verifyTxRelayEon(uint64 relayEon_) {
+        require(relayEon == relayEon_, "Tx doesn't belong to current relayEon");
         _;
     }
 
-    modifier canPause() {
-        require(hasRole(RELAYER_ROLE, msg.sender) || _isOwner() || hasRole(DELEGATE_ROLE, msg.sender), "Only relayer, admin or delegate");
+    modifier canPause(uint256 pauseSinceBlockNumber) {
+        if (pauseSinceBlockNumber > block.number) // Checking UN-pausing (the most critical operation)
+        {
+            require(_isOwner(), "Must have admin role");
+        }
+        else
+        {
+            require(hasRole(MONITOR_ROLE, msg.sender) || _isOwner(), "Must have admin or monitor role");
+        }
         _;
     }
 
-    modifier verifyTxRelayEon(uint64 relayEon_) {
-        require(relayEon == relayEon_, "Tx doesn't belong to current relayEon");
+    modifier canSetReverseAggregatedAllowance(uint256 allowance) {
+        if (allowance > reverseAggregatedAllowanceApproverCap) // Check for going over the approver cap (the most critical operation)
```

Comment:
Approver can set allowance only up to its cap

Reply:
Yes, I put this in very intentionally. Reason being, that apporver is **HOT** wallet (prov key will be available in approver service) = potentially it can get exposed. Approver service will be responsible for automatic increase of reverse aggr. allowance when it start to run low **IF** approver service evaluates that everything is OK. This cap jusy limits ho much allowance can approver (hot wallet) do.